### PR TITLE
feat: [CDS-41423]: nexus2 artifact support

### PR DESCRIPTION
**Pull request by shivamnegi94**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/36093)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 1 Issues 
* Code issue 1 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/36093)

File: 127-cd-nextgen-entities/src/main/java/io/harness/cdng/artifact/bean/yaml/NexusRegistryArtifactConfig.java [hunk 1]
```diff
@@ -66,28 +71,18 @@ public class NexusRegistryArtifactConfig implements ArtifactConfig, Visitable, W
    * Repo name.
    */
   @NotNull @ApiModelProperty(dataType = SwaggerConstants.STRING_CLASSPATH) @Wither ParameterField<String> repository;
-  /**
```
Comment (puthrayaharness) [inline 127-cd-nextgen-entities/src/main/java/io/harness/cdng/artifact/bean/yaml/NexusRegistryArtifactConfig.java:69]: We should not remove this. We need to make the YAML backwards compatible. We should give priority to new changes. If new changes are not present then we should pick up old fields.

**Review by puthrayaharness — CHANGES_REQUESTED** [review]

**Review by puthrayaharness — APPROVED** [review]

**Review by puthrayaharness — APPROVED** [review]
> Reviewed 1 of 9 files at r1, 14 of 29 files at r2, 4 of 18 files at r3, 49 of 49 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36093)* status: :shipit: complete! ✓

**Review by sahilhindwani-harness — APPROVED** [review]
> Reviewed 4 of 49 files at r4.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36093)* status: :shipit: complete! ✓

**Comment by shivamnegi94** [comment]
trigger messagecheck

**Review by puthrayaharness — APPROVED** [review]
> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36093)* status: :shipit: complete! ✓

**Review by hextechpal — APPROVED** [review]
> Reviewed 1 of 9 files at r1, 14 of 29 files at r2, 4 of 18 files at r3, 48 of 49 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36093)* status: :shipit: complete! ✓

**Pull request merged**